server/src/main/java/org/apache/iotdb/db/service/IoTDB.java

```diff
@@ -97,6 +98,7 @@ private void setUp() throws StartupException {
     registerManager.register(Measurement.INSTANCE);
     registerManager.register(SyncServerManager.getInstance());
     registerManager.register(TVListAllocator.getInstance());
+    registerManager.register(MergeManager.getINSTANCE());
```

Comment:
I think that registering MergeManager need before registering StorageEngine, because you execute RecoverMergeTask in StorageEngine.

Reply:
fixed

Reply:
Actually, it's okay since RecoverMergeTasks are not submitted to the MergeManager, they are done synchronously instead.